libgnucash/app-utils/gnc-quotes.cpp

```diff
@@ -146,37 +174,89 @@ static const std::string empty_string{};
+
+        // Check for 0 exit code and exactly one line from wrapper
+        if (rv || info.size() != 1)
+        {
+            std::string err{bl::translate("Failed to initialize Finance::Quote: ")};
+            for (const auto& err_line : errors)
+                err += err_line.empty() ? "" : err_line + "\n";
+            throw(GncQuoteSourceError(err));
+        }
+        // errors should be empty too
+        if (!errors.empty())
+        {
+            std::string err{bl::translate("Finance::Quote check returned error ")};
+            for(const auto& err_line : errors)
+                err += err.empty() ? "" : err_line + "\n";
+            throw(GncQuoteSourceError(err));
+        }
+
+        // Parse JSON
+        try 
+        {
+            std::istringstream is(info.front());
+            bpt::ptree root;
+            bpt::read_json(is, root);
+
+            // Get version and remove it from the tree
+            auto version = root.get<std::string>("version");
+            root.erase("version");
+
+            // Remaining entries look like: key : {key : [value, ...], ...}
+            // Parse as KeyMap, which is string -> ListMap (string -> vec)
+            KeyMap info;
+            for (auto& [outer_k, outer_v]: root) {
+                auto& outer_entry = info[outer_k];
+                for (auto& [inner_k, inner_v]: outer_v) {
+                    auto& inner_entry = outer_entry[inner_k];
+                    for (auto& value : inner_v) {
+                        inner_entry.push_back(value.second.data());
+                    }
+                }
+            }
+
+            // info["currency_modules"] = { "name" : ["feature_name", ...] }
+            // info["quote_methods"]    = { "name" : ["module_name", ...] }
+            // info["quote_modules"]    = { "name" : ["feature_name", ...] }
+            //
+            // A quote_method is the name for one or more quote sources, such as nasdaq (multiple) or alphavantage (single)
+            // A quote_module is backs one or more methods and may have required features.
+            // A "feature_name" is something like "API_KEY", a need value for the source
+
+            m_version = std::move(version);
+
+            for (auto& it : info["currency_modules"]) {
+                m_currency_sources.push_back(it.first);
+            }
+            std::sort (m_currency_sources.begin(), m_currency_sources.end());
+            
+            for (auto& it : info["quote_methods"]) {
+                // Sort methods into single vs multiple sources
+                if (info["quote_methods"][it.first].size() == 1)
+                    m_single_sources.push_back(it.first);
+                else
+                    m_multiple_sources.push_back(it.first);
+            }
+            std::sort (m_single_sources.begin(), m_single_sources.end());
+            std::sort (m_multiple_sources.begin(), m_multiple_sources.end());
```

Comment:
ISTM you've created a lot of complexity by maintaining these as separate containers. Why not a single container of `std::pair<gnc_quote_type, std::string>` and do the separation only once in `gnc_quote_source_add_new()`?